utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/AvroDefaultExpressionExtractor.scala

```diff
@@ -0,0 +1,117 @@
+      toExpression(fieldSchema.schema(), fieldSchema.defaultVal())
+    else
+      Valid(None)
+
+  /**
+    * The expression has to correspond to type extracted by AvroSchemaTypeDefinitionExtractor#typeDefinition
+    * !when applying changes keep in mind that this Schema.Type pattern matching is duplicated in AvroSchemaTypeDefinitionExtractor
+    */
+  @tailrec
+  private def toExpression(schema: Schema, defaultValue: AnyRef): ValidatedNel[AvroDefaultToSpELExpressionError, Option[Expression]] = {
+    schema.getType match {
+      case Schema.Type.RECORD =>
+        typeNotSupported
+      case Schema.Type.ENUM =>
+        withValidation[String](str => s"'$str'")
+      case Schema.Type.ARRAY =>
+        typeNotSupported
+      case Schema.Type.MAP =>
+        typeNotSupported
+      case Schema.Type.UNION =>
+        schema.getTypes.asScala.toList.map(s => (s, s.getType)) match {
+          case (_, Schema.Type.NULL) :: (unionSchemaPart, _) :: Nil =>
+            toExpression(unionSchemaPart, defaultValue)
+          case (unionSchemaPart, _) :: (_, Schema.Type.NULL) :: Nil =>
+            toExpression(unionSchemaPart, defaultValue)
+          case _ =>
+            typeNotSupported
+        }
+      case Schema.Type.STRING if schema.getLogicalType == LogicalTypes.uuid() =>
+        typeNotSupported
+      case Schema.Type.BYTES | Schema.Type.FIXED if schema.getLogicalType != null && schema.getLogicalType.isInstanceOf[LogicalTypes.Decimal] =>
+        typeNotSupported
+      case Schema.Type.STRING =>
+        withValidation[String](str => s"'$str'")
+      case Schema.Type.BYTES =>
+        typeNotSupported
+      case Schema.Type.FIXED =>
+        typeNotSupported
+      case Schema.Type.INT if schema.getLogicalType == LogicalTypes.date() =>
+        typeNotSupported
+      case Schema.Type.INT if schema.getLogicalType == LogicalTypes.timeMillis() =>
+        typeNotSupported
+      case Schema.Type.INT =>
+        withValidation[Integer](_.toString)
+      case Schema.Type.LONG if schema.getLogicalType == LogicalTypes.timestampMillis() || schema.getLogicalType == LogicalTypes.timestampMicros() =>
+        typeNotSupported
+      case Schema.Type.LONG if schema.getLogicalType == LogicalTypes.timeMicros() =>
+        typeNotSupported
+      case Schema.Type.LONG =>
+        withValidation[java.lang.Long](l => s"${l}L")
+      case Schema.Type.FLOAT =>
+        withValidation[java.lang.Float](_.toString)
+      case Schema.Type.DOUBLE =>
+        withValidation[java.lang.Double](_.toString)
+      case Schema.Type.BOOLEAN =>
+        withValidation[java.lang.Boolean](_.toString)
+      case Schema.Type.NULL =>
+        validatedNullExpression
+    }
+  }
+
+  private def withValidation[T <: AnyRef : ClassTag](toExpression: T => Expression): ValidatedNel[AvroDefaultToSpELExpressionError, Option[Expression]] =
+    Option(fieldSchema.defaultVal()) match {
+      case Some(JsonProperties.NULL_VALUE) => validatedNullExpression
```

Comment:
avro is parsing default value from schema with jackson library, and it allows the default to be of the type of the first type in union.
When type is [null | string] the json has to be:
{ "default": null }
otherwise avro type exception is thrown during schema parsing.

so when union is [string | null ]
this: { "default": null } is invalid
this {"default": "null" } will be treatet as string.